test/e2e/framework/framework.go

```diff
@@ -208,6 +208,45 @@ func waitForPodsReady(client v1client.CoreV1Interface, timeout time.Duration, ex
 	}
 }
 
+func (f *Framework) WaitForServiceReady(timeout time.Duration, serviceName string) error {
```

Comment:
@brancz You mean something like a function "ticker" that takes a function with the specific logic?